src/main/java/com/hotels/hcommon/ssh/SshSettings.java

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright (C) 2018 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.hcommon.ssh;
+
+import static com.hotels.hcommon.ssh.validation.Preconditions.checkArgument;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+
+import com.hotels.hcommon.ssh.validation.constraint.TunnelRoute;
+import com.hotels.hcommon.ssh.validation.validator.TunnelRouteValidator;
+
+// TODO choose between validation annotation and preconditions
```

Comment:
Do you want to do this now or leaving this TODO for future?

Reply:
I'm going to remove the TODO, we can decide later